omisego-sdk/src/main/java/co/omisego/omisego/websocket/OMGSocketClient.kt

```diff
@@ -0,0 +1,264 @@
+        wsClient = null
+    }
+
+    /**
+     * A [OMGSocketClient.Builder] used to define the required data to create an instance of the [OMGSocketClient]
+     */
+    class Builder(init: Builder.() -> Unit) : SocketClientContract.Builder {
+
+        /**
+         * (Required) A client configuration that need to be first initialized before calling build()
+         */
+        override var clientConfiguration: ClientConfiguration? = null
+
+        /**
+         * (Optional) A boolean indicating if debug info should be printed in the console. Default: false.
+         */
+        override var debug: Boolean = false
+
+        /**
+         * Create a [OMGSocketClient] instance to be used for connecting to the web socket API.
+         *
+         * @return An instance of the [OMGSocketClient].
+         * @throws IllegalStateException if [authenticationToken], [apiKey] or the [baseURL] is empty.
+         */
+        override fun build(): SocketClientContract.Client {
+            val (baseURL, apiKey, authenticationToken) = clientConfiguration
+                ?: throw IllegalStateException(Exceptions.MSG_NULL_CLIENT_CONFIGURATION)
+
+            check(authenticationToken.isNotEmpty()) { Exceptions.MSG_EMPTY_AUTH_TOKEN }
+            check(apiKey.isNotEmpty()) { Exceptions.MSG_EMPTY_API_KEY }
+            check(baseURL.isNotEmpty()) { Exceptions.MSG_EMPTY_BASE_URL }
+
+            val request = Request.Builder().apply {
+                url(baseURL)
+                addHeader(HTTPHeaders.AUTHORIZATION, "${HTTPHeaders.AUTHORIZATION_SCHEME} ${OMGEncryption.createAuthorizationHeader(apiKey, authenticationToken)}")
+                addHeader(HTTPHeaders.ACCEPT, HTTPHeaders.ACCEPT_OMG)
+            }.build()
+
+            val okHttpClient = OkHttpClient.Builder().apply {
+                /* If set debug true, then print the http logging */
+                if (debug) {
+                    addInterceptor(HttpLoggingInterceptor().apply {
+                        level = HttpLoggingInterceptor.Level.BODY
+                    })
+                }
+            }.build()
+
+            val gson = GsonProvider.create()
+
+            val socketClient = OMGSocketClient(
+                okHttpClient,
+                request,
+                SocketSendParser(gson)
+            )
+
+            val socketDelegator = SocketDelegator(SocketReceiveParser(gson))
+            val socketDispatcher = SocketDispatcher(socketDelegator, SystemEventDispatcher(), CustomEventDispatcher())
+            socketDelegator talksTo socketDispatcher
+
+            val socketChannel = SocketChannel(socketDispatcher, socketClient)
+            socketClient talksTo socketChannel
+            socketDispatcher talksTo socketChannel
+
+            socketClient.wsClient = null
+
+            return socketClient
+        }
+
+        init {
+            init()
+        }
+    }
+}
+
+infix fun OMGSocketClient.talksTo(socketChannel: SocketClientContract.Channel) {
```

Comment:
Any reason why you chose to extract the `talksTo` methods outside of the class body?

Reply:
Indeed, not sure it's a good idea, if it's applied on a class and in the same file than the class declaration, it probably belongs inside the class declaration 🤔 
Or maybe you could regroup all of them in one file, it would be even easier to see which class `talksTo` another class? I think this would be a nice middle ground? Plus, you'll keep the nice coloration we get by making it an extension method 😂  (Not sure if you can change intellij's scheme to use colors on prefix methods always)